docs/messaging/kafka.md

```diff
@@ -127,41 +127,50 @@ If an intermediary broker is present, `service.name` and `peer.service` will not
 
 ### Apache Kafka with Quarkus or Spring Boot Example
 
-Given is a process P, that publishes a message to a topic T1 on Apache Kafka.
-One process, CA, receives the message and publishes a new message to a topic T2 that is then received and processed by CB.
-
-Frameworks such as Quarkus and Spring Boot separate processing of a received message from producing subsequent messages out.
-For this reason, receiving (Span Rcv1) is the parent of both processing (Span Proc1) and producing a new message (Span Prod2).
-The span representing message receiving (Span Rcv1) should set `messaging.operation.type` to `process`,
-as it does not only receive the message but also converts the input message to something suitable for the processing operation to consume and creates the output message from the result of processing.
-
-```
-Process P:  | Span Prod1 |
---
-Process CA:              | Span Rcv1 |
-                                | Span Proc1 |
-                                  | Span Prod2 |
---
-Process CB:                           | Span Rcv2 |
+In this example, the producer publishes a message to a topic T on Apache Kafka.
+Consumer receives the message, processes it and commits the offset.
+
+Frameworks such as Quarkus and Spring Boot provide integrations with Kafka allowing to
+configure and instrument processing callbacks, so corresponding instrumentations should create "Process"
+spans in addition to "Receive" spans created by Kafka instrumentations for polling calls.
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  P[Span 'send T']
+  end
+  subgraph CONSUMER
+  direction TB
+  R1[Span 'poll T']
+  R2[Span 'process T']
+  R3[Span 'commit T']
```

Comment:
```suggestion
  P[Span Send T]
  end
  subgraph CONSUMER
  direction TB
  R1[Span Poll T]
  R2[Span Process T]
  R3[Span Commit T]
```

That would be consistent with other examples. No strong opinion which approach to take, however, ideally we should be consistent.